product-scenarios/1-user-self-registration-to-an-application/1.1-user-registration-with-web-application-itself/1.1.1-provision-user-using-SCIM2/src/test/java/org/wso2/identity/scenarios/test/scim2/ProvisionUserSCIM2TestCase.java

```diff
@@ -0,0 +1,161 @@
+
+        StringEntity entity = new StringEntity(rootObject.toString());
+        request.setEntity(entity);
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 201, "User has not been created successfully");
+
+        Object responseObj = JSONValue.parse(EntityUtils.toString(response.getEntity()));
+        EntityUtils.consume(response.getEntity());
+
+        String usernameFromResponse = ((JSONObject) responseObj).get(USER_NAME_ATTRIBUTE).toString();
+        assertEquals(usernameFromResponse, USERNAME);
+
+        userId = ((JSONObject) responseObj).get(ID_ATTRIBUTE).toString();
+        assertNotNull(userId);
+
+        testDeleteUser();
+    }
+
+    private void testDeleteUser() throws Exception {
+
+        String userResourcePath = getIdentityHTTPSEP() + SCIM2_USERS_ENDPOINT + "/" + userId;
+        HttpDelete request = new HttpDelete(userResourcePath);
+        request.addHeader(HttpHeaders.AUTHORIZATION, getAuthzHeader());
+        request.addHeader(HttpHeaders.CONTENT_TYPE, "application/json");
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 204, "User has not been retrieved successfully");
+        EntityUtils.consume(response.getEntity());
+        userResourcePath = getIdentityHTTPSEP() + SCIM2_USERS_ENDPOINT + "/" + userId;
+        HttpGet getRequest = new HttpGet(userResourcePath);
+        getRequest.addHeader(HttpHeaders.AUTHORIZATION, getAuthzHeader());
+        getRequest.addHeader(HttpHeaders.CONTENT_TYPE, "application/json");
+
+        response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 404, "User has not been deleted successfully");
+        EntityUtils.consume(response.getEntity());
+    }
+
+    private String getAuthzHeader() {
+
+        return "Basic " + Base64.encodeBase64String((ADMIN_USERNAME + ":" + ADMIN_PASSWORD).getBytes()).trim();
+    }
+
+    private void setKeyStoreProperties() {
+
+        System.setProperty("javax.net.ssl.trustStore", resourceLocation + "keystores/products/wso2carbon.jks");
+        System.setProperty("javax.net.ssl.trustStorePassword", "wso2carbon");
+        System.setProperty("javax.net.ssl.trustStoreType", "JKS");
+    }
+
+    private String getIdentityHTTPSEP() {
+
+        String bucketLocation = System.getenv("DATA_BUCKET_LOCATION");
+        String url = null;
+        log.info("Data Bucket location is set : " + bucketLocation);
+
+        Properties prop = new Properties();
+        //InputStream input = null;
+        try (InputStream input = new FileInputStream(bucketLocation + "/infrastructure.properties")) {
+            prop.load(input);
+            url = prop.getProperty("ISHttpsUrl");
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+        if (url == null) {
+            url = "https://localhost:9443";
```

Comment:
use constants to build the default IS url

Reply:
Updated using the base class